eng/scripts/Publish-MavenPackages.ps1

```diff
@@ -48,13 +48,19 @@ if ($packageDetails.Length -eq 0) {
   throw "Aborting, no packages to publish."
 }
 
+if($StageOnly)
+{
+  foreach ($packageDetail in $packageDetails) {
+    if ($packageDetail.IsSnapshot) {
+      throw "Package $($packageDetail.FullyQualifiedName) is a Snapshot and StageOnly is set to 'true'. Staging of snapshot packages is not supported."
+    }
+  }
+}
+
 Write-Host "Starting GPG signing and publishing"
 
 foreach ($packageDetail in $packageDetails) {
   Write-Host "GPG signing and publishing package: $($packageDetail.FullyQualifiedName)"
-  $localRepositoryDirectory = Get-RandomRepositoryDirectory
```

Comment:
All of the options that only apply to non-snapshot maven central packages are moved to the maven central block